From: Michael Froman <mjfroman@mac.com>
Date: Wed, 7 May 2025 15:18:21 -0500
Subject: (tmp-cherry-pick) Revert "Remove cricket::MediaType as a separate
 enum definition" (75abbe9c93)

This reverts commit 886c99a5ac7147b8228cd94204dde077fb1fab86.

Reason for revert: breaks downstream project

Original change's description:
> Remove cricket::MediaType as a separate enum definition
>
> Using constants and alias names to ensure that existing code compiles.
> The part that is not backwards compatible is switch statements that
> don't have default; they will react to the new value ANY.
>
> Bug: webrtc:42222911
> Change-Id: I83320381d8f1b0acbf1adba3ac5c04bcd17e3d6f
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/381660
> Commit-Queue: Harald Alvestrand <hta@webrtc.org>
> Reviewed-by: Evan Shrubsole <eshr@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#44164}

Bug: webrtc:42222911
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Change-Id: Ifffddb374b6fd93cb151c40d62d645f863f0d2c1
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/381820
Reviewed-by: Harald Alvestrand <hta@webrtc.org>
Auto-Submit: Fredrik Solenberg <solenberg@webrtc.org>
Commit-Queue: Harald Alvestrand <hta@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#44166}
---
 api/media_types.cc                            | 12 ++---
 api/media_types.h                             | 47 +++++--------------
 audio/channel_receive.cc                      |  2 +-
 audio/channel_send.cc                         |  2 +-
 pc/peer_connection_factory.cc                 |  8 +++-
 pc/rtc_stats_collector_unittest.cc            |  8 ++--
 pc/sdp_offer_answer.cc                        |  3 --
 pc/webrtc_sdp.cc                              | 23 +++++----
 sdk/android/src/jni/pc/media_stream_track.cc  |  2 +-
 .../peerconnection/RTCRtpCodecCapability.mm   |  5 +-
 .../peerconnection/RTCRtpCodecParameters.mm   |  5 +-
 sdk/objc/api/peerconnection/RTCRtpReceiver.h  |  1 -
 sdk/objc/api/peerconnection/RTCRtpReceiver.mm | 26 ++++------
 13 files changed, 61 insertions(+), 83 deletions(-)

diff --git a/api/media_types.cc b/api/media_types.cc
index 9cd45324b4..850c8e2d35 100644
--- a/api/media_types.cc
+++ b/api/media_types.cc
@@ -14,7 +14,7 @@
 
 #include "rtc_base/checks.h"
 
-namespace webrtc {
+namespace cricket {
 
 const char kMediaTypeVideo[] = "video";
 const char kMediaTypeAudio[] = "audio";
@@ -22,13 +22,13 @@ const char kMediaTypeData[] = "data";
 
 std::string MediaTypeToString(MediaType type) {
   switch (type) {
-    case MediaType::AUDIO:
+    case MEDIA_TYPE_AUDIO:
       return kMediaTypeAudio;
-    case MediaType::VIDEO:
+    case MEDIA_TYPE_VIDEO:
       return kMediaTypeVideo;
-    case MediaType::DATA:
+    case MEDIA_TYPE_DATA:
       return kMediaTypeData;
-    default:
+    case MEDIA_TYPE_UNSUPPORTED:
       // Unsupported media stores the m=<mediatype> differently.
       RTC_DCHECK_NOTREACHED();
       return "";
@@ -36,4 +36,4 @@ std::string MediaTypeToString(MediaType type) {
   RTC_CHECK_NOTREACHED();
 }
 
-}  // namespace webrtc
+}  // namespace cricket
diff --git a/api/media_types.h b/api/media_types.h
index 339c26398e..430bab5965 100644
--- a/api/media_types.h
+++ b/api/media_types.h
@@ -17,30 +17,7 @@
 
 namespace webrtc {
 
-enum class MediaType {
-  AUDIO,
-  VIDEO,
-  DATA,
-  UNSUPPORTED,
-  ANY,
-  // Backwards compatibility values for cricket::MediaType users
-  // TODO: https://issues.webrtc.org/42222911 - deprecate and remove
-  MEDIA_TYPE_AUDIO = AUDIO,
-  MEDIA_TYPE_VIDEO = VIDEO,
-  MEDIA_TYPE_DATA = DATA,
-  MEDIA_TYPE_UNSUPPORTED = UNSUPPORTED,
-};
-
-RTC_EXPORT std::string MediaTypeToString(MediaType type);
-
-template <typename Sink>
-void AbslStringify(Sink& sink, MediaType type) {
-  sink.Append(MediaTypeToString(type));
-}
-
-extern const char kMediaTypeAudio[];
-extern const char kMediaTypeVideo[];
-extern const char kMediaTypeData[];
+enum class MediaType { AUDIO, VIDEO, DATA, UNSUPPORTED, ANY };
 
 }  // namespace webrtc
 
@@ -50,18 +27,18 @@ extern const char kMediaTypeData[];
 
 namespace cricket {
 
-using MediaType = webrtc::MediaType;
-using webrtc::kMediaTypeAudio;
-using webrtc::kMediaTypeData;
-using webrtc::kMediaTypeVideo;
-using webrtc::MediaTypeToString;
+enum MediaType {
+  MEDIA_TYPE_AUDIO = static_cast<int>(webrtc::MediaType::AUDIO),
+  MEDIA_TYPE_VIDEO = static_cast<int>(webrtc::MediaType::VIDEO),
+  MEDIA_TYPE_DATA = static_cast<int>(webrtc::MediaType::DATA),
+  MEDIA_TYPE_UNSUPPORTED = static_cast<int>(webrtc::MediaType::UNSUPPORTED),
+};
 
-// Backwards compatibility values for cricket::MediaType users
-// TODO: https://issues.webrtc.org/42222911 - deprecate and remove
-constexpr MediaType MEDIA_TYPE_AUDIO = webrtc::MediaType::AUDIO;
-constexpr MediaType MEDIA_TYPE_VIDEO = webrtc::MediaType::VIDEO;
-constexpr MediaType MEDIA_TYPE_DATA = webrtc::MediaType::DATA;
-constexpr MediaType MEDIA_TYPE_UNSUPPORTED = webrtc::MediaType::UNSUPPORTED;
+extern const char kMediaTypeAudio[];
+extern const char kMediaTypeVideo[];
+extern const char kMediaTypeData[];
+
+RTC_EXPORT std::string MediaTypeToString(MediaType type);
 
 }  // namespace cricket
 
diff --git a/audio/channel_receive.cc b/audio/channel_receive.cc
index 2d7a0b9fc3..349e919b4c 100644
--- a/audio/channel_receive.cc
+++ b/audio/channel_receive.cc
@@ -751,7 +751,7 @@ void ChannelReceive::ReceivePacket(const uint8_t* packet,
     char buf[1024];
     SimpleStringBuilder mime_type(buf);
     auto it = payload_type_map_.find(header.payloadType);
-    mime_type << cricket::MediaTypeToString(cricket::MEDIA_TYPE_AUDIO) << "/"
+    mime_type << MediaTypeToString(cricket::MEDIA_TYPE_AUDIO) << "/"
               << (it != payload_type_map_.end() ? it->second.name
                                                 : "x-unknown");
     frame_transformer_delegate_->Transform(payload_data, header, remote_ssrc_,
diff --git a/audio/channel_send.cc b/audio/channel_send.cc
index 01cf8a6daf..75f512fb26 100644
--- a/audio/channel_send.cc
+++ b/audio/channel_send.cc
@@ -385,7 +385,7 @@ int32_t ChannelSend::SendData(AudioFrameType frameType,
     // is transformed, the delegate will call SendRtpAudio to send it.
     char buf[1024];
     SimpleStringBuilder mime_type(buf);
-    mime_type << cricket::MediaTypeToString(cricket::MEDIA_TYPE_AUDIO) << "/"
+    mime_type << MediaTypeToString(cricket::MEDIA_TYPE_AUDIO) << "/"
               << encoder_format_.name;
     frame_transformer_delegate_->Transform(
         frameType, payloadType, rtp_timestamp + rtp_rtcp_->StartTimestamp(),
diff --git a/pc/peer_connection_factory.cc b/pc/peer_connection_factory.cc
index 2cb54d262d..849323c706 100644
--- a/pc/peer_connection_factory.cc
+++ b/pc/peer_connection_factory.cc
@@ -166,7 +166,9 @@ RtpCapabilities PeerConnectionFactory::GetRtpSenderCapabilities(
           GetDefaultEnabledRtpHeaderExtensions(media_engine()->video());
       return ToRtpCapabilities(cricket_codecs, extensions);
     }
-    default:
+    case cricket::MEDIA_TYPE_DATA:
+      return RtpCapabilities();
+    case cricket::MEDIA_TYPE_UNSUPPORTED:
       return RtpCapabilities();
   }
   RTC_DLOG(LS_ERROR) << "Got unexpected MediaType " << kind;
@@ -191,7 +193,9 @@ RtpCapabilities PeerConnectionFactory::GetRtpReceiverCapabilities(
           GetDefaultEnabledRtpHeaderExtensions(media_engine()->video());
       return ToRtpCapabilities(cricket_codecs, extensions);
     }
-    default:
+    case cricket::MEDIA_TYPE_DATA:
+      return RtpCapabilities();
+    case cricket::MEDIA_TYPE_UNSUPPORTED:
       return RtpCapabilities();
   }
   RTC_DLOG(LS_ERROR) << "Got unexpected MediaType " << kind;
diff --git a/pc/rtc_stats_collector_unittest.cc b/pc/rtc_stats_collector_unittest.cc
index 61268c4d31..2aa66e723e 100644
--- a/pc/rtc_stats_collector_unittest.cc
+++ b/pc/rtc_stats_collector_unittest.cc
@@ -3222,7 +3222,8 @@ class RTCStatsCollectorTestWithParamKind
         return "A";
       case cricket::MEDIA_TYPE_VIDEO:
         return "V";
-      default:
+      case cricket::MEDIA_TYPE_DATA:
+      case cricket::MEDIA_TYPE_UNSUPPORTED:
         RTC_DCHECK_NOTREACHED();
         return "?";
     }
@@ -3234,7 +3235,8 @@ class RTCStatsCollectorTestWithParamKind
         return "audio";
       case cricket::MEDIA_TYPE_VIDEO:
         return "video";
-      default:
+      case cricket::MEDIA_TYPE_DATA:
+      case cricket::MEDIA_TYPE_UNSUPPORTED:
         RTC_DCHECK_NOTREACHED();
         return "";
     }
@@ -3283,7 +3285,7 @@ class RTCStatsCollectorTestWithParamKind
         return;
       }
       case cricket::MEDIA_TYPE_DATA:
-      default:
+      case cricket::MEDIA_TYPE_UNSUPPORTED:
         RTC_DCHECK_NOTREACHED();
     }
   }
diff --git a/pc/sdp_offer_answer.cc b/pc/sdp_offer_answer.cc
index 8986e921a6..13a5d393b7 100644
--- a/pc/sdp_offer_answer.cc
+++ b/pc/sdp_offer_answer.cc
@@ -731,9 +731,6 @@ absl::string_view GetDefaultMidForPlanB(cricket::MediaType media_type) {
       return cricket::CN_DATA;
     case cricket::MEDIA_TYPE_UNSUPPORTED:
       return "not supported";
-    default:
-      // Fall through to RTC_CHECK_NOTREACHED
-      break;
   }
   RTC_DCHECK_NOTREACHED();
   return "";
diff --git a/pc/webrtc_sdp.cc b/pc/webrtc_sdp.cc
index ba47e756a2..c4a24c3aef 100644
--- a/pc/webrtc_sdp.cc
+++ b/pc/webrtc_sdp.cc
@@ -43,6 +43,7 @@
 #include "media/base/rtp_utils.h"
 #include "media/base/stream_params.h"
 #include "media/sctp/sctp_transport_internal.h"
+#include "p2p/base/candidate_pair_interface.h"
 #include "p2p/base/ice_transport_internal.h"
 #include "p2p/base/p2p_constants.h"
 #include "p2p/base/port.h"
@@ -235,9 +236,9 @@ static const char kAttrGroup[] = "a=group:BUNDLE";
 static const char kConnectionNettype[] = "IN";
 static const char kConnectionIpv4Addrtype[] = "IP4";
 static const char kConnectionIpv6Addrtype[] = "IP6";
-static const char kSdpMediaTypeVideo[] = "video";
-static const char kSdpMediaTypeAudio[] = "audio";
-static const char kSdpMediaTypeData[] = "application";
+static const char kMediaTypeVideo[] = "video";
+static const char kMediaTypeAudio[] = "audio";
+static const char kMediaTypeData[] = "application";
 static const char kMediaPortRejected[] = "0";
 // draft-ietf-mmusic-trickle-ice-01
 // When no candidates have been gathered, set the connection
@@ -1413,14 +1414,14 @@ void BuildMediaLine(const cricket::MediaType media_type,
   std::string fmt;
   if (media_type == cricket::MEDIA_TYPE_AUDIO ||
       media_type == cricket::MEDIA_TYPE_VIDEO) {
-    type = media_type == cricket::MEDIA_TYPE_AUDIO ? kSdpMediaTypeAudio
-                                                   : kSdpMediaTypeVideo;
+    type = media_type == cricket::MEDIA_TYPE_AUDIO ? kMediaTypeAudio
+                                                   : kMediaTypeVideo;
     for (const cricket::Codec& codec : media_desc->codecs()) {
       fmt.append(" ");
       fmt.append(rtc::ToString(codec.id));
     }
   } else if (media_type == cricket::MEDIA_TYPE_DATA) {
-    type = kSdpMediaTypeData;
+    type = kMediaTypeData;
     const cricket::SctpDataContentDescription* sctp_data_desc =
         media_desc->as_sctp();
     if (sctp_data_desc) {
@@ -2716,23 +2717,21 @@ bool ParseMediaDescription(
     bool bundle_only = false;
     int section_msid_signaling = cricket::kMsidSignalingNotUsed;
     absl::string_view media_type = fields[0];
-    if ((media_type == kSdpMediaTypeVideo ||
-         media_type == kSdpMediaTypeAudio) &&
+    if ((media_type == kMediaTypeVideo || media_type == kMediaTypeAudio) &&
         !cricket::IsRtpProtocol(protocol)) {
       return ParseFailed(*mline, "Unsupported protocol for media type", error);
     }
-    if (media_type == kSdpMediaTypeVideo) {
+    if (media_type == kMediaTypeVideo) {
       content = ParseContentDescription(
           message, cricket::MEDIA_TYPE_VIDEO, mline_index, protocol,
           payload_types, pos, &content_name, &bundle_only,
           &section_msid_signaling, &transport, candidates, error);
-    } else if (media_type == kSdpMediaTypeAudio) {
+    } else if (media_type == kMediaTypeAudio) {
       content = ParseContentDescription(
           message, cricket::MEDIA_TYPE_AUDIO, mline_index, protocol,
           payload_types, pos, &content_name, &bundle_only,
           &section_msid_signaling, &transport, candidates, error);
-    } else if (media_type == kSdpMediaTypeData &&
-               cricket::IsDtlsSctp(protocol)) {
+    } else if (media_type == kMediaTypeData && cricket::IsDtlsSctp(protocol)) {
       // The draft-03 format is:
       // m=application <port> DTLS/SCTP <sctp-port>...
       // use_sctpmap should be false.
diff --git a/sdk/android/src/jni/pc/media_stream_track.cc b/sdk/android/src/jni/pc/media_stream_track.cc
index 297bbf67bc..225b608e1c 100644
--- a/sdk/android/src/jni/pc/media_stream_track.cc
+++ b/sdk/android/src/jni/pc/media_stream_track.cc
@@ -21,7 +21,7 @@ namespace jni {
 ScopedJavaLocalRef<jobject> NativeToJavaMediaType(
     JNIEnv* jni,
     cricket::MediaType media_type) {
-  return Java_MediaType_fromNativeIndex(jni, static_cast<int>(media_type));
+  return Java_MediaType_fromNativeIndex(jni, media_type);
 }
 
 cricket::MediaType JavaToNativeMediaType(JNIEnv* jni,
diff --git a/sdk/objc/api/peerconnection/RTCRtpCodecCapability.mm b/sdk/objc/api/peerconnection/RTCRtpCodecCapability.mm
index e9726943de..f6f72e859b 100644
--- a/sdk/objc/api/peerconnection/RTCRtpCodecCapability.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpCodecCapability.mm
@@ -47,7 +47,10 @@
       case cricket::MEDIA_TYPE_VIDEO:
         _kind = kRTCMediaStreamTrackKindVideo;
         break;
-      default:
+      case cricket::MEDIA_TYPE_DATA:
+        RTC_DCHECK_NOTREACHED();
+        break;
+      case cricket::MEDIA_TYPE_UNSUPPORTED:
         RTC_DCHECK_NOTREACHED();
         break;
     }
diff --git a/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm b/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm
index 579607ba6b..fa69c36cdd 100644
--- a/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm
@@ -59,7 +59,10 @@ const NSString *const kRTCH264CodecName = @(cricket::kH264CodecName);
       case cricket::MEDIA_TYPE_VIDEO:
         _kind = kRTCMediaStreamTrackKindVideo;
         break;
-      default:
+      case cricket::MEDIA_TYPE_DATA:
+        RTC_DCHECK_NOTREACHED();
+        break;
+      case cricket::MEDIA_TYPE_UNSUPPORTED:
         RTC_DCHECK_NOTREACHED();
         break;
     }
diff --git a/sdk/objc/api/peerconnection/RTCRtpReceiver.h b/sdk/objc/api/peerconnection/RTCRtpReceiver.h
index 2877a4c12b..d0c79fece7 100644
--- a/sdk/objc/api/peerconnection/RTCRtpReceiver.h
+++ b/sdk/objc/api/peerconnection/RTCRtpReceiver.h
@@ -22,7 +22,6 @@ typedef NS_ENUM(NSInteger, RTCRtpMediaType) {
   RTCRtpMediaTypeVideo,
   RTCRtpMediaTypeData,
   RTCRtpMediaTypeUnsupported,
-  RTCRtpMediaTypeAny,
 };
 
 @class RTC_OBJC_TYPE(RTCRtpReceiver);
diff --git a/sdk/objc/api/peerconnection/RTCRtpReceiver.mm b/sdk/objc/api/peerconnection/RTCRtpReceiver.mm
index 3f2dd10c51..bb5b01d16b 100644
--- a/sdk/objc/api/peerconnection/RTCRtpReceiver.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpReceiver.mm
@@ -140,33 +140,29 @@ void RtpReceiverDelegateAdapter::OnFirstPacketReceived(
 }
 
 + (RTCRtpMediaType)mediaTypeForNativeMediaType:
-    (webrtc::MediaType)nativeMediaType {
+    (cricket::MediaType)nativeMediaType {
   switch (nativeMediaType) {
-    case webrtc::MediaType::AUDIO:
+    case cricket::MEDIA_TYPE_AUDIO:
       return RTCRtpMediaTypeAudio;
-    case webrtc::MediaType::VIDEO:
+    case cricket::MEDIA_TYPE_VIDEO:
       return RTCRtpMediaTypeVideo;
-    case webrtc::MediaType::DATA:
+    case cricket::MEDIA_TYPE_DATA:
       return RTCRtpMediaTypeData;
-    case webrtc::MediaType::UNSUPPORTED:
+    case cricket::MEDIA_TYPE_UNSUPPORTED:
       return RTCRtpMediaTypeUnsupported;
-    case webrtc::MediaType::ANY:
-      return RTCRtpMediaTypeAny;
   }
 }
 
-+ (webrtc::MediaType)nativeMediaTypeForMediaType:(RTCRtpMediaType)mediaType {
++ (cricket::MediaType)nativeMediaTypeForMediaType:(RTCRtpMediaType)mediaType {
   switch (mediaType) {
     case RTCRtpMediaTypeAudio:
-      return webrtc::MediaType::AUDIO;
+      return cricket::MEDIA_TYPE_AUDIO;
     case RTCRtpMediaTypeVideo:
-      return webrtc::MediaType::VIDEO;
+      return cricket::MEDIA_TYPE_VIDEO;
     case RTCRtpMediaTypeData:
-      return webrtc::MediaType::DATA;
+      return cricket::MEDIA_TYPE_DATA;
     case RTCRtpMediaTypeUnsupported:
-      return webrtc::MediaType::UNSUPPORTED;
-    case RTCRtpMediaTypeAny:
-      return webrtc::MediaType::ANY;
+      return cricket::MEDIA_TYPE_UNSUPPORTED;
   }
 }
 
@@ -180,8 +176,6 @@ void RtpReceiverDelegateAdapter::OnFirstPacketReceived(
       return @"DATA";
     case RTCRtpMediaTypeUnsupported:
       return @"UNSUPPORTED";
-    case RTCRtpMediaTypeAny:
-      return @"ANY";
   }
 }
 
